test/adapter/json_api/relationship_test.rb

```diff
@@ -35,6 +35,20 @@ def test_relationship_with_nil_model
           assert_equal(expected, actual)
         end
 
+        def test_relationship_with_nil_model_and_belongs_to_id_on_self
+          ActiveModelSerializers.config.jsonapi_use_foreign_key_on_belongs_to_relationship = true
```

Comment:
I would prefer to save the original config before setting it to our use case (`true` in this case) and restore it later in the `ensure` block
```rb
  original_config = ActiveModelSerializers.config.jsonapi_use_foreign_key_on_belongs_to_relationship
  ActiveModelSerializers.config.jsonapi_use_foreign_key_on_belongs_to_relationship = true

  # lines 41-49
ensure
  ActiveModelSerializers.config.jsonapi_use_foreign_key_on_belongs_to_relationship = original_config
end
```